pkg/deviceshifu/deviceshifuopcua/deviceshifuopcua.go

```diff
@@ -103,14 +102,14 @@ func New(deviceShifuMetadata *deviceshifubase.DeviceShifuMetaData) (*DeviceShifu
 						opcua.AuthCertificate(cert.Certificate[0]),
 					)
 				case ua.UserTokenTypeUserName:
-					passwordByte, err := os.ReadFile(DeviceSecretPasswordPath)
+					password, exist := base.DeviceShifuSecret[OPCUASettingPasswordSecret]
 					// secret will overwrite the password in edge device
-					if err != nil {
-						klog.Infof("secret load error: %v, password will be loaded from OPCUASetting.Password", err)
-						options = append(options, opcua.AuthUsername(*setting.Username, *setting.Password))
-					} else {
+					if exist {
 						klog.Infof("password loaded from secret")
-						options = append(options, opcua.AuthUsername(*setting.Username, string(passwordByte)))
+						options = append(options, opcua.AuthUsername(*setting.Username, password))
+					} else {
+						klog.Infof("secret not found, password will be loaded from OPCUASetting.Password")
+						options = append(options, opcua.AuthUsername(*setting.Username, *setting.Password))
```

Comment:
Does deviceshifu crash when a user does not enter both a secret and a password?

Reply:
Yes. `setting.Password` will be nil.